File: leetcode_75/Sliding_Window/cpp/643.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution
{
public:
    double findMaxAverage(vector<int> &nums, int k)
    {
        int cur_s = 0;
        for (int i = 0; i < k; ++i)
        {
            cur_s += nums[i];
        }
        int mx_sum = cur_s;

        for (int i = k; i < nums.size(); ++i)
        {
            cur_s += nums[i] - nums[i - k];
            mx_sum = max(mx_sum, cur_s);
        }

        
        return (double)mx_sum / k;
    }
};

File: leetcode_75/Sliding_Window/python/643.py
from typing import List


class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        cur_s = 0
        for i in range(k):
            cur_s += nums[i]
        mx_sum = cur_s

        for i in range(k, len(nums)):
            cur_s += nums[i] - nums[i - k]
            mx_sum = max(mx_sum, cur_s)

        return mx_sum / k
